platform/linux-generic/pktio/pcap.c

```diff
@@ -276,6 +278,16 @@ static int pcapif_recv_pkt(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 			break;
 		}
 
+		if (pktio_cls_enabled(pktio_entry)) {
+			ret = cls_classify_packet(pktio_entry, data,
+						  pkt_len, PACKET_PARSE_SEG_LEN,
+						  &new_pool, pkt_hdr, true);
+			if (ret) {
+				odp_packet_free(pkt);
+				continue;
+			}
+		}
+
 		packet_parse_layer(pkt_hdr,
```

Comment:
`cls_classify_packet()` performs full packet parse, so `packet_parse_layer()` call becomes unnecessary when classifier is enabled.